### PR TITLE
fix: Refactor tabs to remove React-Router dependency 

### PR DESCRIPTION
**Pull request by jeffredodd**
### Description
In order to remove the dependency on React Router from Fundamentals-React (#115), we must refactor the the following components...

- Breadcrumb (#294)
- Menu (#293)
- Tabs (#336)
- SideNavigation (Coming Soon)

This PR is focused on **Tab** and does introduce breaking changes such as...

- Refactoring of JSX components
- `<TabComponent>` now encapsulates `<Tab>` 
- `<Tab>` no longer creates its own `<Link>` but instead allows it to be added via an HOC pattern. 

Since we needed to support Tabs ingesting `<Link>` via a higher order component, there was a major change to the `<JSX>` API. I'm open to renaming any component, just give a shoutout!

**Tab Component**
```
<TabComponent>
        <Tab
            content='Hello world'
            id='1'
            url='#'>
            Tab 1
        </Tab>
        <Tab
            content='Hello world 2'
            id='2'
            url='#'>
            Tab 2
        </Tab>
        <Tab
            content='Hello world 3'
            disabled
            id='3'
            url='#'>
            Tab 3
        </Tab>
    </TabComponent>
```

**Tab Component w/ Anchors**
```
 <TabComponent>
        <Tab
            content='Hello world'
            id='1'>
            <a
                href='#'>
                Tab 1
            </a>
        </Tab>
        <Tab
            content='Hello world 2'
            id='2'>
            <a
                href='#'>
                Tab 2
            </a>
        </Tab>
        <Tab
            content='Hello world 3'
            disabled
            id='3'>
            <a
                href='#'>
                Tab 2
            </a>
        </Tab>
    </TabComponent>
```

**Tab Component w/ `<Link>`**
```
<MemoryRouter>
        <TabComponent>
            <Tab
                content='Hello world'
                id='1'>
                <Link
                    to='/'>
                    Tab 1
                </Link>
            </Tab>
            <Tab
                content='Hello world 2'
                id='2'>
                <Link
                    to='/'>
                    Tab 2
                </Link>
            </Tab>
            <Tab
                content='Hello world 3'
                disabled
                id='3'>
                <Link
                    to='/'>
                    Tab 2
                </Link>
            </Tab>
        </TabComponent>
    </MemoryRouter>
```

### CODE REVIEW

File: src/_playground/documentation/Playground/Playground.js [hunk 1]
```diff
@@ -434,9 +434,9 @@ export class Playground extends Component {
               break;
           case 'tabs':
               componentToGenerate = (
-                  <Tabs>
+                  <Tab>
                       <TabComponent ids={this.state.childs.children.props.ids} />
-                  </Tabs>
+                  </Tab>
```
[the remark this answers is not in the document]
Comment (jeffredodd) [reply src/_playground/documentation/Playground/Playground.js]: Missed this, this was a playground example that we are no longer using. I've moved this for now.

**Comment by jeffredodd** [comment]
@greg-a-smith, all feedback should be up to date now 😄

File: src/Tabs/Tabs.js [hunk 2]
```diff
@@ -1,111 +1,129 @@
 import classnames from 'classnames';
 import PropTypes from 'prop-types';
-import { BrowserRouter, Link } from 'react-router-dom';
 import React, { Component } from 'react';
 
-export const Tabs = props => {
-    const { children, className, ...rest } = props;
+export const Tab = props => {
+    const { children, className, content, disabled, id, onClick, selectedId, tabContentProps, tabLinkProps, url, ...rest } = props;
```
Comment (cjskillingstad) [inline src/Tabs/Tabs.js]: Looks like some of these are missing in the `propTypes` definition. This could be an argument to use classes rather than SFC's.

File: src/Tabs/Tabs.js [hunk 3]
```diff
@@ -1,111 +1,129 @@
...
+        if (url) {
+            return (
+                <a
+                    {...tabLinkProps}
+                    aria-disabled={disabled}
+                    className={className}
+                    disabled={disabled}
+                    href={url}
+                    onClick={(e) => {
+                        if (!disabled && onClick) {
```
Comment (cjskillingstad) [inline src/Tabs/Tabs.js]: If we default `onClick: () => {}` we won't have to check if its defined.

File: src/Tabs/Tabs.js [hunk 4]
```diff
@@ -1,111 +1,129 @@
...
+                </a>
+            );
+        } else if (children && React.isValidElement(children)) {
+            return React.cloneElement(children, {
+                ...tabLinkProps,
+                'aria-disabled': disabled,
+                className: className,
+                disabled: disabled,
+                onClick: (e) => {
+                    if (!disabled && onClick) {
```
Comment (cjskillingstad) [inline src/Tabs/Tabs.js]: Same as above.

File: src/Tabs/Tabs.js (hunk 2 above)
Comment (greg-a-smith) [inline src/Tabs/Tabs.js]: It looks like there are a number of these props that are not defined in the `propTypes` or `propDescriptions`.  The `Tab` component is only showing `id`, `children`, `className` and `url`.

File: src/Tabs/Tabs.js [hunk 5]
```diff
@@ -1,111 +1,129 @@
...
-        'fd-tabs-container',
-        className
-    );
+    const renderLink = () => {
+        if (url) {
+            return (
+                <a
+                    {...tabLinkProps}
+                    aria-disabled={disabled}
+                    className={className}
```
Comment (greg-a-smith) [inline src/Tabs/Tabs.js]: Does the `className` prop go here or should that go on the `<li>`?  If they really wanted a `className` on the `<a>` or `Link`, they could pass it through `tabLinkProps`.

File: src/Tabs/Tabs.js [hunk 6]
```diff
@@ -1,111 +1,129 @@
...
+                        }
+                    }}>
+                    {children}
+                </a>
+            );
+        } else if (children && React.isValidElement(children)) {
+            return React.cloneElement(children, {
+                ...tabLinkProps,
+                'aria-disabled': disabled,
+                className: className,
```
Comment (greg-a-smith) [inline src/Tabs/Tabs.js]: Same as the comment above on the `<a>` for when `url` is passed.

File: src/Tabs/Tabs.js [hunk 7]
```diff
@@ -1,111 +1,129 @@
...
             }
         );
     }
 
+    renderTabs = (children) => {
+        return React.Children.map(children, (child) => {
+            const classes = this.getLinkClasses(child.props.id, child.props.className);
+
+            return React.cloneElement(child, {
+                selectedId: this.state.selectedId,
```
Comment (greg-a-smith) [inline src/Tabs/Tabs.js]: Instead of passing the `selectedId` to each child `Tab`, what if the comparison of `child.props.id` to `selectedId` occurred here and the `Tab` component was simply passed a `selected` prop?  Keep more of the decision making in this component (that is carrying state) and leave the `Tab` as a dumb component.

File: src/Tabs/Tabs.js [hunk 8]
```diff
@@ -1,111 +1,129 @@
...
 };
 
-TabComponent.propDescriptions = {
-    ids: 'An array of objects with keys \'id\', \'url\', \'name\', \'hasChild\', \'child\', and \'glyph\' defining each tab.',
-    tabContentProps: 'Additional props to be spread to the tab\'s content.',
-    tabLinkProps: 'Additional props to be spread to the tab\'s link.',
-    tabProps: 'Additional props to be spread to the tab.'
+TabGroup.propDescriptions = {
+    children: 'One or more Tab components to render within the component.',
+    selectedId: 'Initial selected `<Tab>`'
```
Comment (greg-a-smith) [inline src/Tabs/Tabs.js]: Maybe reword to "The `id` of the selected `Tab`."

File: src/Tabs/Tabs.js [hunk 9]
```diff
@@ -1,111 +1,129 @@
...
+    selectedId: PropTypes.string
 };
 
-TabComponent.propDescriptions = {
-    ids: 'An array of objects with keys \'id\', \'url\', \'name\', \'hasChild\', \'child\', and \'glyph\' defining each tab.',
-    tabContentProps: 'Additional props to be spread to the tab\'s content.',
-    tabLinkProps: 'Additional props to be spread to the tab\'s link.',
-    tabProps: 'Additional props to be spread to the tab.'
+TabGroup.propDescriptions = {
+    children: 'One or more Tab components to render within the component.',
```
Comment (greg-a-smith) [inline src/Tabs/Tabs.js]: "Tab" should be backticked.

**Comment by greg-a-smith** [comment]
@jeffredodd There is an issue (#315) where the tab content appears misaligned for Tab 2.  I am seeing this in the documentation examples.  Is this an easy fix that could be added to this?

File: src/Tabs/Tabs.js (hunk 5 above)
Comment (jeffredodd) [reply src/Tabs/Tabs.js]: This functionality was not present in the last version of tabs so lets put it on the `<li>` and if a user needs it that can add it to `tabLinkProps` to get it to the anchor.

**Comment by jeffredodd** [comment]
@greg-a-smith So when doing research it appeared that the [Fundamental Tabs](https://sap.github.io/fundamental/components/tabs.html) don't currently document how they support content. 

The jittering I think we are observing is due to the content of a tab being loaded into the `<li>` with a `<p>` which is not the right place for it. I didn't change it in this PR as I was trying to alter as little outputted markup as possible.
